### ISSUE TITLE
Step 24: Forcing Mechanisms

### ISSUE THREAD

**Issue by anjor** [anjor/gandalf#29]
Implement forcing for driven turbulence studies (Thesis §2.5.1):

1. **Create krmhd/forcing.py** with:
   - `gaussian_white_noise(k_force, amplitude, dt, key)`: JAX random noise
   - `force_alfven_modes(state, k_force, amplitude, dt, key)`
   - `force_slow_modes(state, k_force, amplitude, dt, key)`

2. **Alfvén sector forcing**:
   - Drive both Elsasser fields (ξ⁺ and ξ⁻) identically
   - This forces u⊥ = ẑ × ∇Φ only, NOT δB⊥
   - **Critical**: Avoids artificial large-scale reconnection
   - Specify forcing wavenumbers (kₓ, kᵧ, kz)

3. **Slow mode forcing** (optional):
   - Force δB∥ or Hermite moments (g₀, g₁) directly
   - Alternative: Force perpendicular current (drives δB⊥ → δB∥)
   - Independent from Alfvén forcing

4. **JAX random keys**:
   - Use `jax.random.split(key)` for reproducibility
   - Thread keys through forcing function
   - Enable deterministic simulations with fixed seed

5. **Test suite**:
   - Measure energy injection rate: dE/dt from forcing
   - Verify Alfvén forcing doesn't drive δB∥ (should remain passive)
   - Check forcing spectrum concentrated at specified k_force

**Dependencies:** Step 8 (Time integrator), Step 17 (Hermite moments)
**Next Step:** Step 25

**Comment by anjor** [comment]
**Status Check** ✓ Still needed - No implementation yet

## Current State
- No `forcing.py` module exists in `src/krmhd/`
- No Gaussian white noise forcing
- No energy injection capability

## Context
Current simulations are **decaying turbulence only** (Issue #12). For **driven turbulence** studies (saturated state, energy balance, steady-state spectra), need external energy injection.

**Current capability**:
- ✅ Decaying turbulence with exponential energy decay
- ❌ Driven turbulence with energy injection = dissipation

## Physics Importance
Critical for:
1. **Steady-state turbulence** - Balance forcing and dissipation
2. **Issue #27** - Kinetic FDT validation requires single-k forcing
3. **Issue #35** - Performance benchmarks on saturated turbulence
4. **Real astrophysics** - Solar wind, accretion disks have continuous energy injection

## Implementation Requirements

**Alfvén sector forcing** (most important):
```python
force_alfven_modes(state, k_force, amplitude, dt, key)
```
- Drive both z± identically → forces u⊥ only, not δB⊥
- Critical: Avoids artificial large-scale reconnection
- Specify forcing wavenumbers (kₓ, kᵧ, kz)

**JAX random keys**:
- Use `jax.random.split(key)` for reproducibility
- Thread keys through forcing function
- Enable deterministic simulations with fixed seed

**Slow mode forcing** (optional):
- Force δB∥ or Hermite moments (g₀, g₁) directly
- Independent from Alfvén forcing
- Less commonly used

## Dependencies
- **Issue #27** (Kinetic FDT) needs single-mode forcing
- Would enhance Issue #12 (Decaying turbulence) → driven turbulence
- Independent of other issues

## Priority
**Medium-High** - Essential for steady-state turbulence studies and kinetic validation (Issue #27).

Keeping open - important for driven turbulence physics. 🎲
